src/stores/GuildMemberRoleStore.js

```diff
@@ -0,0 +1,192 @@
+      if (!this._roles.includes(roleOrRoles.id)) return Promise.resolve(this.member);
+      return this.client.api.guilds(this.guild.id).members(this.member.id).roles(roleOrRoles.id)
+        .delete({ reason })
+        .then(() => {
+          const clone = this.member._clone();
+          const roles = clone.roles._roles;
+
+          const index = roles.indexOf(roleOrRoles.id);
+          if (~index) roles.splice(index, 1);
+
+          clone.roles._patch(roles);
+          return clone;
+        });
+    } else {
+      const allRoles = this._roles.slice();
+      for (let role of roleOrRoles instanceof Collection ? roleOrRoles.values() : roleOrRoles) {
+        role = this.resolve(role);
+        if (!role) {
+          return Promise.reject(new TypeError('INVALID_TYPE', 'roles',
+            'Array or Collection of Roles or Snowflakes', true));
+        }
+        const index = allRoles.indexOf(role.id);
+        if (index >= 0) allRoles.splice(index, 1);
+      }
+      return this.member.edit({ roles: allRoles }, reason);
+    }
+  }
+
+  /**
+   * The role of the member used to hoist them in a separate category in the users list
+   * @type {?Role}
+   * @readonly
+   */
+  get hoistRole() {
+    const hoistedRoles = this.filter(role => role.hoist);
+    if (!hoistedRoles.size) return null;
+    return hoistedRoles.reduce((prev, role) => !prev || role.comparePositionTo(prev) > 0 ? role : prev);
+  }
+
+  /**
+   * The role of the member used to set their color
+   * @type {?Role}
+   * @readonly
+   */
+  get colorRole() {
+    const coloredRoles = this.filter(role => role.color);
+    if (!coloredRoles.size) return null;
+    return coloredRoles.reduce((prev, role) => !prev || role.comparePositionTo(prev) > 0 ? role : prev);
+  }
+
+  /**
+   * The role of the member with the highest position
+   * @type {Role}
+   * @readonly
+   */
+  get highestRole() {
+    return this.reduce((prev, role) => !prev || role.comparePositionTo(prev) > 0 ? role : prev);
+  }
+
+  resolve(idOrInstance) {
+    return this.guild.roles.resolve(idOrInstance);
+  }
+
+  resolveID(idOrInstance) {
+    return this.guild.roles.resolveID(idOrInstance);
+  }
+
+  /**
+   * Patches the roles for this store
+   * @param {Snowflake[]} roles The new roles
+   * @private
+   */
+  _patch(roles) {
+    this._roles = roles;
+
+    this.clear();
+
+    const everyoneRole = this.guild.roles.get(this.guild.id);
+    if (everyoneRole) super.set(everyoneRole.id, everyoneRole);
+
+    if (roles) {
+      for (const roleID of roles) {
+        const role = this.resolve(roleID);
+        if (role) super.set(role.id, role);
+      }
+    }
+  }
+
+  /**
+   * Resolves a RoleResolvable to a Role object.
+   * @method resolve
+   * @memberof GuildMemberRoleStore
+   * @instance
+   * @param {RoleResolvable} role The role resolvable to resolve
+   * @returns {?Role}
+   */
```

Comment:
hopefully you remove the method override, in which case this would be correct.  however, as it stands currently, please move this to the relevant location above the method it's documenting.